File: baekjoon_js/dfs&bfs/2331.js
("use strict");

const fs = require("fs");
const stdin = (process.platform === "linux"
  ? fs.readFileSync("/dev/stdin").toString()
  : `57 2
`
).split("\n");

const input = (() => {
  let line = 0;
  return () => stdin[line++];
})();

let [A, P] = input().split(" ").map(Number);

let numList = [A];
let nextA = 0;
while (true) {
  for (let j = 0; j < String(A).length; j++) {
    number = Math.pow(String(A)[j], P);
    nextA += number;
  }
  //   console.log(nextA);
  if (numList.find((e) => e === nextA) === undefined) numList.push(nextA);
  else {
    console.log(numList.findIndex((e) => e === nextA));
    break;
  }
  A = nextA;
  nextA = 0;
}

File: baekjoon/dfs&bfs/2331.py



A, P = map(int,input().split())

numList = []
isVisited = []
numList.append(A)
nextA = 0

while True:
    strA = str(A)
    endPoint = -1
    nextA = 0
    for i in range(len(strA)):
        nextA += int(strA[i]) ** P
    for i in range(len(numList)):
        if nextA == numList[i]:
            endPoint = i
            break
    if endPoint > -1:
        break
    else:
        numList.append(nextA)
    A = nextA
print(len(numList[:endPoint]))

print( 9 ** 5)